### PR TITLE
RDoc-770 delete-by-query-operation

### PR DESCRIPTION
**Pull request by IdanHaim**
- change SizeUnit.Bytes to SizeUnit.Megabytes in SetupAggressiveCaching
- fix Note tag in how-to/delete-documents-using-index-with-linq.markdown

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿# Operations: How to delete documents by query
+
+The `DeleteByQueryOperation` gives you the ability to delete a large amount of documents with single Quary.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: typo: Query -> query

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+## Example
+{CODE-TABS}
+{CODE-TAB:csharp:Sync delete_by_query2@ClientApi\Operations\DeleteByQuery.cs /}
+{CODE-TAB:csharp:Async delete_by_query3@ClientApi\Operations\DeleteByQuery.cs /}
+{CODE-TAB-BLOCK/}
+{CODE-TABS/}
+
+
+#### Remarks
+{NOTE: Importent}
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: typo: importent -> important

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+                using (var session = store.OpenSession())
+                {
+                    #region delete_by_query2
+
+                    // remove all documents from the server where Name == Bob using Person/ByName index
+                    var operation1 =
+                        store.Operations.Send(
+                            new DeleteByQueryOperation<Person>("Person/ByName", x => x.Name == "Bob"));
+                    operation1.WaitForCompletion(TimeSpan.FromSeconds(15));
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: That needs an explanation in the article that DeleteByQueryOperation is performed in the background on the server and you have an _option_ to wait for it using `WaitForCompletion`.

The examples suggest that it's mandatory

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+                    //remove all documents from the server where Age > 35 using Person/ByAge index
+                    var operation2 =
+                        store.Operations.Send(
+                            new DeleteByQueryOperation<Person, Person_ByAge>(x => x.Age < 35));
+                    operation2.WaitForCompletion(TimeSpan.FromSeconds(15));
+
+                    // remove all document form Pepole collection in the server
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: typos: document form Pepole -> documents form People

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+| **options** | QueryOperationOptions | Holds different setting options for base operations |
+
+
+
+## Example
+{CODE-TABS}
+{CODE-TAB:csharp:Sync delete_by_query2@ClientApi\Operations\DeleteByQuery.cs /}
+{CODE-TAB:csharp:Async delete_by_query3@ClientApi\Operations\DeleteByQuery.cs /}
+{CODE-TAB-BLOCK/}
+{CODE-TABS/}
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: RQL syntax is missing in the examples. Other articles typically have separate {CODE-TABS} for each example so RQL is put for each of those.

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            }
+        }
+
+        public async Task Sample()
+        {
+            using (var store = new DocumentStore())
+            {
+                store.Initialize();
+
+                using (var session = store.OpenSession())
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: rename `session` to `asyncSession` to have a user the hint it's async

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+                            new DeleteByQueryOperation<Person>("Person/ByName", x => x.Name == "Bob"));
+                    operation1.WaitForCompletion(TimeSpan.FromSeconds(15));
+
+                    //remove all documents from the server where Age > 35 using Person/ByAge index
+                    var operation2 =
+                        await store.Operations.SendAsync(
+                            new DeleteByQueryOperation<Person, Person_ByAge>(x => x.Age < 35));
+                    operation2.WaitForCompletion(TimeSpan.FromSeconds(15));
+
+                    // remove all document form Pepole collection in the server
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: the same typos as above

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 4 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: typo in the typo :)

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown (hunk 1 above)
Comment (IdanHaim) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown (hunk 2 above)
Comment (IdanHaim) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 3 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: You are right will be fix

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 4 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 6 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: deleted this now need for session

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 7 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: fix

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 4 above)
Comment (arekpalinski) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: :)

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 6 above)
Comment (arekpalinski) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: why deleted?

**Comment by arekpalinski** [comment]
Please also add http://issues.hibernatingrhinos.com/issue/RDoc-668 to this article

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs (hunk 6 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: This operation is used in the store not in the session

Comment (arekpalinski) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/DeleteByQuery.cs]: right

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown (hunk 5 above)
Comment (IdanHaim) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/client/delete-by-query-operation.markdown]: done

**Pull request closed without merging**